components/binary_sensor/xiaomi.py

```diff
@@ -150,39 +121,35 @@ def is_on(self):
         """Return true if sensor is on."""
         return self._state
 
+    @property
+    def device_state_attributes(self):
+        attrs = {ATTR_OPEN_SINCE: self._open_since}
+        attrs.update(super().device_state_attributes)
+        return attrs
+
     def parse_data(self, data):
-        if not 'status' in data:
+        if NO_CLOSE in data:  # handle push from the hub
+            self._open_since = data[NO_CLOSE]
+            return True
+
+        value = data.get(self._data_key)
+        if value is None:
             return False
 
-        value = data['status']
         if value == 'open' or value == 'no_close':

```

Comment:
Probably yes. homeassistant.const has "open", but nothing for close. Maybe for the completeness we should have both defined in this file though?